controllers/progressivesync_controller.go

```diff
@@ -224,7 +230,7 @@ func (r *ProgressiveSyncReconciler) Reconcile(ctx context.Context, req ctrl.Requ
 // SetupWithManager adds the reconciler to the manager, so that it gets started when the manager is started.
 func (r *ProgressiveSyncReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&syncv1alpha1.ProgressiveSync{}).
+		For(&syncv1alpha1.ProgressiveSync{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
```

Comment:
See: https://github.com/operator-framework/operator-sdk/issues/2795

This effectively makes the reconciler not reconcile on its own status changes.

Reply:
While this unblock us for the time being, I'd like to raise a separate issue to have a long term fix for this because now the controller can't self-heal if something changes the status.

Reply:
A +1 to raising this as a separate issue.

Reply:
Done: https://github.com/Skyscanner/applicationset-progressive-sync/issues/99